pkg/csi/service/wcp/controller_helper.go

```diff
@@ -330,34 +335,40 @@ func getDatastoreURLFromStoragePool(ctx context.Context, spName string) (string,
 	return datastoreURL, nil
 }
 
-// getAccessibleNodesFromStoragePool returns the accessibleNodes pertaining to the given StoragePool
-func getAccessibleNodesFromStoragePool(ctx context.Context, spName string) ([]string, error) {
+// getStoragePoolInfo returns the accessibleNodes and the storage-pool-type pertaining to the given StoragePool
+func getStoragePoolInfo(ctx context.Context, spName string) ([]string, string, error) {
 	// Get a config to talk to the apiserver
 	cfg, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
+		return nil, "", fmt.Errorf("failed to get Kubernetes config. Err: %+v", err)
 	}
 
 	// create a new StoragePool client
 	spClient, err := dynamic.NewForConfig(cfg)
```

Comment:
@vMansiShah spClient here is a K8S client which isn't CO agnostic.

Reply:
Since this change does not add or alter this functionality we can take this discussion offline to decide if this code needs to change. But resolving the comment since it is not about my code changes